src/Aspire.Dashboard/Otlp/WebApplicationExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using Aspire.Dashboard.Authentication;
+using Aspire.Dashboard.Otlp.Grpc;
+using Microsoft.AspNetCore.Mvc;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace Aspire.Dashboard.Otlp;
+
+public static class WebApplicationExtensions
+{
+    // https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/large-object-heap
+    private const int MaxSizeLessThanLOH = 84999;
+
+    public static WebApplication ConfigureHttpOtlp(this WebApplication app, Uri httpEndpoint)
+    {
+        app.MapPost("/v1/metrics",
+                async ([FromServices] OtlpMetricsService service, HttpContext httpContext) =>
+                {
+                    return await ExportOtlpData(httpContext,
+                            sequence => service.Export(ExportMetricsServiceRequest.Parser.ParseFrom(sequence), null!))
+                        .ConfigureAwait(false);
+                }
+            ).RequireAuthorization(OtlpAuthorization.PolicyName)
+            .RequireHost($"*:{httpEndpoint.Port}");
```

Comment:
I used connection middleware to ensure OTLP + gRPC request came in on the right endpoint. I'd like OTLP + HTTP to be consistent.

There are probably some relatively modest changes/improvements to reuse the same code for both.

Reply:
@JamesNK any idea how to make this work elegantly ?